src/Vector.cpp

```diff
@@ -0,0 +1,335 @@
+#include <Vector.h>
+#include <stdexcept>
+
+// ======================== c-tors & d-tor =======================
+
+/**
+ * ctor creates a vector with a given size and value
+ * @param size the size of the vector
+ * @param value the value to initialize the vector with
+ **/
+Vector::Vector(size_t size, int value):
+	m_size(size),
+	m_capacity(size),
+	m_vector(new int[size])
+{
+	for (size_t i = 0; i < m_size; ++i) {
+		m_vector[i] = value; 
+	}
+}
+
+/**
+ * copy ctor creates a vector from another vector
+ * @param other the vector to copy
+ **/
+Vector::Vector(const Vector& other):
+	m_size(other.m_size),
+	m_capacity(other.m_capacity),
+	m_vector(new int[m_capacity])
+{
+	for (size_t i = 0; i < m_size && i < m_capacity; ++i) {
+		m_vector[i] = other.m_vector[i];
+	}
+}
+
+/**
+ * moving ctor move a vector to another vector
+ * @param other the vector to move
+ **/
+Vector::Vector(Vector&& other) noexcept:
+	m_size(other.m_size),
+	m_capacity(other.m_capacity),
+	m_vector(other.m_vector)
+{
+	// prevent access of the moved vector to the memory
+	other.m_vector = nullptr; 
+	other.m_size = other.m_capacity = 0;
+}
+
+/**
+ * d-tor deletes the vector and frees the memory
+ **/
+Vector::~Vector()
+{
+	delete[] m_vector;
+	m_vector = nullptr;
+}
+
+// ======================== methods ==============================
+
+int* Vector::data()
+{
+	return m_vector;
+}
+
+const int* Vector::data() const
+{
+	return m_vector;
+}
+
+bool Vector::empty() const
+{
+	return m_size == 0;
+}
+
+size_t Vector::size() const
+{
+	return m_size;
+}
+
+size_t Vector::capacity() const
+{
+	return m_capacity;
+}
+
+/**
+ * clear the vector and free the memory
+ **/
+void Vector::clear()
+{
+	delete[] m_vector;
```

Comment:
clear should leave the vector's capacity as it was. 
Check this out: https://en.cppreference.com/w/cpp/container/vector/clear

Reply:
fixed